tardis/plasma/tests/test_complete_plasmas.py

```diff
@@ -1 +1,122 @@
+import os
 
+import pandas as pd
+import pytest
+from pandas.util import testing as pdt
+from numpy.testing import assert_almost_equal
+from tardis.io.config_reader import Configuration
+from tardis.simulation import Simulation
+
+
+class BasePlasmaTest(object):
+    #Class defining all common tests for different setups of Plasma
+    #This can then be inherited for different Plasma setup
+
+    @pytest.fixture(scope="class", autouse=True)
+    def plasma(self, atomic_data_fname, tardis_ref_path, reference_file_path, config_path):
+        self.config = Configuration.from_yaml(config_path)
+        self.config['atom_data'] = atomic_data_fname
+        self.sim = Simulation.from_config(self.config)
+
+        if pytest.config.getvalue("--generate-reference"):
+            if os.path.exists(reference_file_path):
+                pytest.skip(
+                    'Reference data {0} does exist and tests will not '
+                    'proceed generating new data'.format(reference_file_path))
+            self.sim.plasma.to_hdf(reference_file_path)
+            pytest.skip("Reference data saved at {0}".format(
+                reference_file_path))
+        return self.sim.plasma
+
+    @pytest.fixture(scope="class")
+    def reference_file_path(self):
+        pass
+
+    @pytest.fixture(scope="class")
+    def config_path(self):
+        pass
+
+    general_properties = ['beta_rad', 'g_electron', 'selected_atoms',
+                          'number_density', 't_electrons', 'w', 't_rad']
+    partiton_properties = ['level_boltzmann_factor', 'partition_function']
+    atomic_properties = ['excitation_energy', 'lines', 'lines_lower_level_index',
+                         'lines_upper_level_index', 'atomic_mass', 'ionization_data',
+                         'nu', 'wavelength_cm', 'f_lu', 'metastability']
+    ion_population_properties = [
+        'phi', 'ion_number_density', 'electron_densities']
+    level_population_properties = ['level_number_density']
+    radiative_properties = ['stimulated_emission_factor',
+                            'tau_sobolevs', 'beta_sobolev', 'transition_probabilities']
+    j_blues_properties = ['j_blues']
+
+    combined_properties = general_properties + partiton_properties + atomic_properties + ion_population_properties + \
+        level_population_properties + radiative_properties + \
+        j_blues_properties
+
+    @pytest.mark.parametrize("attr", combined_properties)
+    def test_plasma_properties(self, reference_file_path, plasma, attr):
+        actual = getattr(plasma, attr)
+        expected = pd.read_hdf(reference_file_path,
```

Comment:
I think we should not use `pd.read_hdf` as I think that opens the file for every test. I think it would be better to have a fixture for the opened `pd.HDFStore`.
I think one could use a 'yield-fixture' for the `pd.HDFStore` (see https://docs.pytest.org/en/2.7.3/yieldfixture.html#yieldfixture)